TagV3.0_U575VGT/Core/Src/Sensor Src/LightSensor.c

```diff
@@ -183,3 +179,34 @@ HAL_StatusTypeDef LightSensor_get_part_id(LightSensorHandleTypedef *light_sensor
 HAL_StatusTypeDef LightSensor_get_manufacturer(LightSensorHandleTypedef *light_sensor, ALSManufacIDRegister *dst){
     return HAL_I2C_Mem_Read(light_sensor->i2c_handler, ALS_ADDR << 1, ALS_MANUFAC_ID_ADDR, I2C_MEMADD_SIZE_8BIT, (uint8_t*)dst, sizeof(ALSManufacIDRegister), 100);
 }
+
+
+void priv_lightSensor_sample_timer_expiration(ULONG light_sensor_raw_addr){
+    LightSensorHandleTypedef *light_sensor = (LightSensorHandleTypedef *)light_sensor_raw_addr;
+	if(LightSensor_get_data(light_sensor) == HAL_OK){
+		//Store data somewhere?
+	}
+
+}
+
+HAL_StatusTypeDef LightSensor_thread_entry(ULONG thread_input){
+
+
+
+	//Declare lightsensor handler and initialize chip
+	LightSensorHandleTypedef *light_sensor = (LightSensorHandleTypedef *)thread_input;
+	HAL_RESULT_PROPAGATE(LightSensor_init(light_sensor, &hi2c2));
```

Comment:
I'm not sure what propagating this result would do since thread entries usually return void. I think its fine for now but it could be worth doing some error handling (in all our sensors not just this one) in the future (e.g., maybe retry the initialization every few seconds).

We could make a click-up task for this and tackle it in the future I think